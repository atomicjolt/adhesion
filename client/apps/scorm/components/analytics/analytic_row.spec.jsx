import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';
import AnalyticRow from './analytic_row';

describe('Scorm Analytics AnalyticRow', () => {

  let result;
  let props;
  describe('first example', () => {
    beforeEach(() => {
      props = {
        id: 1,
        name: 'test application',
        passed: 'Passed',
        score: 0.12,
        time: 59,
        isParent: true,
        show: true,
        depth: 2,
        tableRowClicked: () => {},
      };

      result = shallow(<AnalyticRow {...props} />);
    });

    it('renders', () => {
      expect(result).toMatchSnapshot();
    });

    it('renders the analytic row with the correct values', () => {
      const tr = result.find('.c-aa-row');
      expect(tr).toBeDefined();
      expect(tr.text()).toContain(props.name);
      expect(tr.text()).toContain(props.passed);
      expect(tr.text()).toContain(props.score * 100);
      expect(tr.text()).toContain(`${props.time}s`);
    });

    it('renders the correct amount of table rows', () => {
      const tr = result.find('tr');
      expect(tr.hasClass('c-aa-row c-aa-accordion is-open')).toBeTruthy();
    });
  });

  describe('second example', () => {
    beforeEach(() => {
      props = {
        id: 1,
        name: 'test application',
        passed: 'Failed',
        score: null,
        time: 112,
        isParent: false,
        show: false,
        depth: 3,
        tableRowClicked: () => {},
      };

      result = shallow(<AnalyticRow {...props} />);
    });

    it('renders the table row with the correct values', () => {
      const tr = result.find('tr');
      expect(tr.text()).toContain('N/A');
      expect(tr.text()).toContain(`${_.ceil(props.time / 60)}m`);
    });

    it('renders the table row with the correct amount of rows', () => {
      const tr = result.find('tr');
      expect(tr.hasClass('c-aa-row')).toBeTruthy();
      expect(tr.hasClass('c-aa-row c-aa-accordion is-open')).toBeFalsy();
    });

    it('renders the icon with no values', () => {
      const icons = result.find('i');
      expect(icons.length).toBe(0);
    });
  });
});
