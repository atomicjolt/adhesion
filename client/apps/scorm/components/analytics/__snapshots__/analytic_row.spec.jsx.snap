// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scorm Analytics AnalyticRow first example renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnalyticRow
    depth={2}
    id={1}
    isParent={true}
    name="test application"
    passed="Passed"
    score={0.12}
    show={true}
    tableRowClicked={[Function]}
    time={59}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          style={
                    Object {
                              "paddingLeft": "8rem",
                            }
          }
>
          <i
                    className="material-icons"
                    style={
                              Object {
                                        "left": "5.3rem",
                                      }
                    }
          >
                    arrow_drop_down
          </i>
          test application
</td>,
        <td>
          Passed
</td>,
        <td>
          12%
</td>,
        <td>
          59s
</td>,
      ],
      "className": "c-aa-row c-aa-accordion is-open",
      "onClick": [Function],
      "style": Object {
        "cursor": "pointer",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="material-icons"
              style={
                            Object {
                                          "left": "5.3rem",
                                        }
              }
>
              arrow_drop_down
</i>,
            "test application",
          ],
          "style": Object {
            "paddingLeft": "8rem",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "arrow_drop_down",
              "className": "material-icons",
              "style": Object {
                "left": "5.3rem",
              },
            },
            "ref": null,
            "rendered": "arrow_drop_down",
            "type": "i",
          },
          "test application",
        ],
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Passed",
        },
        "ref": null,
        "rendered": "Passed",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "12%",
        },
        "ref": null,
        "rendered": "12%",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "59s",
        },
        "ref": null,
        "rendered": "59s",
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            style={
                        Object {
                                    "paddingLeft": "8rem",
                                  }
            }
>
            <i
                        className="material-icons"
                        style={
                                    Object {
                                                "left": "5.3rem",
                                              }
                        }
            >
                        arrow_drop_down
            </i>
            test application
</td>,
          <td>
            Passed
</td>,
          <td>
            12%
</td>,
          <td>
            59s
</td>,
        ],
        "className": "c-aa-row c-aa-accordion is-open",
        "onClick": [Function],
        "style": Object {
          "cursor": "pointer",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons"
                style={
                                Object {
                                                "left": "5.3rem",
                                              }
                }
>
                arrow_drop_down
</i>,
              "test application",
            ],
            "style": Object {
              "paddingLeft": "8rem",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "arrow_drop_down",
                "className": "material-icons",
                "style": Object {
                  "left": "5.3rem",
                },
              },
              "ref": null,
              "rendered": "arrow_drop_down",
              "type": "i",
            },
            "test application",
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Passed",
          },
          "ref": null,
          "rendered": "Passed",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "12%",
          },
          "ref": null,
          "rendered": "12%",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "59s",
          },
          "ref": null,
          "rendered": "59s",
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
